app/labor/bufferizer.rb

```diff
@@ -21,21 +21,22 @@ def satellite_tweet!
       if text.length < 250 && SiteConfig.twitter_hashtag
         text = text.gsub(" #{SiteConfig.twitter_hashtag}", " #{SiteConfig.twitter_hashtag} ##{tag.name}")
       end
-      BufferUpdate.buff!(@article.id, text, tag.buffer_profile_id_code, "twitter", tag.id, @admin_id)
+      BufferUpdate.buff!(@article.id, text, tag.buffer_profile_id_code, { tag_id: tag.id, admin_id: @admin_id })
     end
     @article.update(last_buffered: Time.current)
   end
 
   def main_tweet!
-    BufferUpdate.buff!(@article.id, twitter_buffer_text, ApplicationConfig["BUFFER_TWITTER_ID"], "twitter", nil,
-                       @admin_id)
+    BufferUpdate.buff!(@article.id, twitter_buffer_text, ApplicationConfig["BUFFER_TWITTER_ID"],
```

Comment:
Considering you're using the `buff!` method on different places I believe you could consider something like:

```ruby
class Bufferizer
  ...
  def facebook_post!
    buffer_params = build_buffer_params(text: fb_buffer_text, social_service_name: "facebook")

    BufferUpdate.buff!(buffer_params)
  end

  def build_buffer_params(text:, social_service_name:)
    BufferUpdate::Params.new(
      @article, @tag, @admin, text, social_service_name
    )
  end
  ...
end
```

Reply:
I've tried to do a `params_builder` but I think it was too complex. It seems like that:

```rb
  def build_buffer_params(text, social_service_name:, tag:)
    @article.id,
    text,
    Hash.new
      .merge({social_service_name: social_service_name}) if social_service_name
      .merge({buffer_profile_id_code: tag.buffer_profile_id_code}) if tag && tag.buffer_profile_id_code
      .merge({tag_id: tag.id}) if tag
      .merge({admin_id: @admin.id}) if @admin
  end
```

What did you think about it?